### DISCUSSION TITLE
snapChanged

### DISCUSSION THREAD

**Discussion by argyleink** [argyleink/ScrollSnapExplainers#1] [General]
## Context: [explainer](https://github.com/argyleink/ScrollSnapExplainers/blob/main/js-snapChanged)

Questions?
Thoughts?
Naming fine?
Risks?

**Comment by majido** [comment]
An event model for scroll-snap is a fairly  popular request. See [this issue](https://github.com/w3c/csswg-drafts/issues/156) from 2016 with 63 👍 . That issue contains a reasonable [starting proposal](https://github.com/w3c/csswg-drafts/issues/156#issuecomment-695085852) with good questions to consider. I suggest considering the snapChanging/snapChanged proposal as part of a single proposal for adding snap events.

I suggest making these non-blocking events. Any blocking snap event will likely get push back as they will interfere with off-thread scrolling.

**Reply by argyleink** [reply]
thanks for the links and tips!

**Comment by argyleink** [comment]
discussion today around this api mentioned having the results be a multi-dimensional array so it's clear it's a collection of snapped items, and may be 1 but could be many more. i should make some visuals or demo's that help explain this!

**Comment by nachtfunke** [comment]
I am not sure if it makes a difference, but I regularly run into the need for JavaScript events and being able to wire in additional control elements, as well as being able to react to interaction in scroll-snap'd contexts would be really, really important. For now, I have to engage with IntersectionObserver, which feels suboptimal. It works but it feels very inelegant.

So, TLDR; expressing my support for the proposed additions.
